File: android/app/src/main/java/com/locationsharing/app/domain/service/MapService.kt
package com.locationsharing.app.domain.service

import com.locationsharing.app.data.models.Location

interface MapService {
    fun initializeMap()
    fun updateUserLocation(location: Location)
    fun updateFriendLocations(friendLocations: Map<String, Location>)
    fun addLocationMarker(friendId: String, location: Location, friendName: String)
    fun removeLocationMarker(friendId: String)
    fun clearAllMarkers()
}

File: ios/LocationSharingApp/LocationSharingApp/Domain/Services/MapService.swift
import Foundation
import MapKit

protocol MapService {
    func initializeMap()
    func updateUserLocation(_ location: Location)
    func updateFriendLocations(_ friendLocations: [String: Location])
    func addLocationMarker(friendId: String, location: Location, friendName: String)
    func removeLocationMarker(friendId: String)
    func clearAllMarkers()
}